### ISSUE TITLE
Review usage of the `error-message` component

### ISSUE THREAD

**Issue by MartinJJones** [alphagov/govuk_publishing_components#3488]
## What
Remove the `error-message` component used in Government Frontend and replace with the `error-message` component from the gem.

## Why
Although they look the same, the components do have different functionality. The form `error-message` component is based off the design system and includes the Accessibility enhancements below:

> To help screen reader users, the error message component includes a hidden ‘Error:’ before the error message. These users will hear, for example, “Error: The date your passport was issued must be in the past”. - https://design-system.service.gov.uk/components/error-message/

```HTML
<span class="govuk-visually-hidden">Error:</span> Please enter your National Insurance Number
```

## Screenshots

| Publishing component | Government Frontend |
| --- | --- |
| <img width="451" alt="error-message-government-frontend" src="https://github.com/alphagov/govuk_publishing_components/assets/28779939/8242faa3-d35e-4b7c-860e-ce014c45b014"> | <img width="451" alt="error-message-government-frontend" src="https://github.com/alphagov/govuk_publishing_components/assets/28779939/603d2aa8-b845-4ee9-89e6-f49ca1fde5ef"> |

**Comment by jon-kirwan** [comment]
It looks like the error message app component is only included in one view (in Government Frontend) [choose_sign_in.html](https://github.com/alphagov/government-frontend/blob/main/app/views/content_items/service_sign_in/_choose_sign_in.html.erb#L32-L34).

From what I can tell the choose sign in page is no longer used since it contains an option for GOV.UK Verify which is [now closed](https://www.gov.uk/government/publications/introducing-govuk-verify/introducing-govuk-verify). Previously, I think, you would get to this page from https://www.gov.uk/log-in-file-self-assessment-tax-return. Now the sign in link takes you user to a page to sign in using Government Gateway.

See: [link to screenshot of sign in page](https://user-images.githubusercontent.com/2445413/33993427-67881b52-e0ce-11e7-872a-b6da35b78ce9.png) copied from https://github.com/alphagov/government-frontend/pull/610.

**Comment by jon-kirwan** [comment]
On further investigation, I've found that the choose sign in page is still in use, although only for one service, https://www.gov.uk/claim-rural-payments/sign-in/prove-identity. From what I've read it seems likely that once services fully shift from Government Gateway to One Login, it will no longer be needed. However, until that happens, it seems like a good idea to replace the component.

More information [here - Slack](https://gds.slack.com/archives/CAB4PSQKW/p1698163612402929).
